recipes/libsecp256k1/meta.yaml

```diff
@@ -0,0 +1,105 @@
+{% set name = "libsecp256k1" %}
+{% set headers = [
+      "secp256k1.h",
+      "secp256k1_preallocated.h",
+      "secp256k1_ecdh.h",
+      "secp256k1_recovery.h",
+      "secp256k1_extrakeys.h",
+      "secp256k1_schnorrsig.h",
+      "secp256k1_ellswift.h"] %}
+{% set version = "0.4.0" %}
+{% set commit = "199d27cea32203b224b208627533c2e813cd3b21" %}
+{% set sha256 = "3ffdeed78c7bebfe6cbd87bbe99d36982e92a6893d1e8b4dd4fdbf35a8811320" %}
+  
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/bitcoin-core/secp256k1/archive/{{ commit }}.tar.gz
+  sha256: {{ sha256 }}
+  patches:
+    - 0001_update_pc.in.patch
+    - 0002_update_src_cmake_for_pc.in.patch
+    - 0003_create_pkgconfig_function.patch
+
+build:
+  number: 0
+  missing_dso_whitelist:
+    - "*/libc.so.6"
```

Comment:
Please post the logging message without libc in the missing_dso_whitelist?

Reply:
You should be seeing something like:

```
2023-11-16T18:21:36.0530842Z    INFO (libtomo,lib/libtomo-gridrec.so): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libc.so.6 found in CDT/compiler package conda-forge::sysroot_linux-64-2.17-h4a8ded7_13
```

i.e. The linker should be linking your objects to libc in our placeholder sysroot package.

Reply:
The log shows something like this instead:
`Making absolute symlink relative (lib/libsecp256k1.so -> libsecp256k1.so.2 :-> libsecp256k1.so.2.1.0)
  ERROR (libsecp256k1,lib/libsecp256k1.so.2.1.0): /lib64/libc.so.6 not found in packages, sysroot(s) nor the missing_dso_whitelist.
.. is this binary repackaging?
`
How to resolve?

Reply:
It's because you put the compiler in requirements/host instead of requirements/build. Build tools go in requirements/build. Deps that are linked to and need to match the target platform when cross-compiling go in requirements/host.

Reply:
I changed deps from pkgconfig (python) to pkg-config (CLI). I'm not sure if that was correct or not.